### PR TITLE
Issue1093 feedback form should have a headline or subject

### PR DESCRIPTION
**Pull request by miguelahonen**
This pull request includes:

- A new subject field in the feedback form.
- The subject of the received email is a combination of a service name and a subject text written by a user.
- WCAG 2.1 AA level improvements (html label tags and aria-labels) in the feedback form.
- Language support (fi, sv, en) for the new subject field, html label tags and aria-labels in the feedback form.

### CODE REVIEW

File: resource/translations/skosmos_sv.po [hunk 1]
```diff
@@ -836,3 +836,12 @@ msgstr "Sökresultat"
...
 msgstr "Sidofältlista: ordna och bläddra begrepp efter kriterier"
+
+msgid "Subject:"
+msgstr "Ämne:"
+
+msgid "Write a subject"
+msgstr "Skriv ett ämne"
+
+msgid "Select a vocabulary"
+msgstr "Du kan välja ett vokabulär"
```
Comment (osma) [inline resource/translations/skosmos_sv.po:847]: I believe this should be "en vokabulär", not "ett vokabulär" (I checked the other Swedish translations)

File: controller/WebController.php [hunk 2]
```diff
@@ -213,18 +213,17 @@ public function invokeFeedbackForm($request)
         }
         $feedbackName = $request->getQueryParamPOST('name');
         $feedbackEmail = $request->getQueryParamPOST('email');
+        $msgSubject = $this->model->getConfig()->getServiceName() . ": ";
```
[the remark this answers is not in the document]
Comment (osma) [reply controller/WebController.php:216]: Also, this is really nitpicking but...since the formatting of the message body is done in the sendFeedback method, I think it would be more consistent to also format the subject line within sendFeedback, not here. So just pass the `msgsubject` POST variable as it is to sendFeedback, then add the prefix within sendFeedback - where the subject line used to be formatted before this change.

**Review by osma — CHANGES_REQUESTED** [review]
Looks very good! I gave a few small comments and requests for possible improvements.

File: controller/WebController.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (miguelahonen) [reply controller/WebController.php:216]: Good ideas - Done!

File: controller/WebController.php [hunk 3]
```diff
@@ -245,46 +244,43 @@ private function createFeedbackHeaders($fromName, $fromEmail, $toMail, $sender)
...
         }
-
         $service = $this->model->getConfig()->getServiceName();
         return $headers . "From: $fromName via $service <$sender>";
     }
 
     /**
      * Sends the user entered message through the php's mailer.
      * @param string $message only required parameter is the actual message.
+     * @param string $messageSubject from the sender.
```
[the remark this answers is not in the document]
Comment (miguelahonen) [reply controller/WebController.php:254]: I changed this

File: controller/WebController.php [hunk 4]
```diff
@@ -245,46 +244,43 @@ private function createFeedbackHeaders($fromName, $fromEmail, $toMail, $sender)
...
             $headers .= "Reply-To: $fromName <$fromEmail>\r\n";
         }
-
         $service = $this->model->getConfig()->getServiceName();
         return $headers . "From: $fromName via $service <$sender>";
     }
 
     /**
      * Sends the user entered message through the php's mailer.
      * @param string $message only required parameter is the actual message.
```
[the remark this answers is not in the document]
Comment (miguelahonen) [reply controller/WebController.php:253]: Done

File: view/feedback.twig [hunk 5]
```diff
@@ -30,15 +30,22 @@
...
+        </p>
+        <p>
+          <label for="msgsubject">{% trans %}Subject:{% endtrans %}</label>
+          <input id="msgsubject" class="form-control" type="text" size="40" name="msgsubject" placeholder="{% trans %}Write a subject{% endtrans %}">
+        </p>
+        <label for="message">{% trans %}Message:{% endtrans %}</label>
         <textarea id="message" class="form-control" name="message"></textarea>
         {{ honeypot.generate('item-description', 'user-captcha')|raw }}
-        <button type="submit" class="btn btn-primary" id="send-feedback">{% trans %}Send feedback{% endtrans %}</button>
+        <button type="submit" class="btn btn-primary" id="send-feedback" aria-label="{% trans %}Send feedback{% endtrans %}">{% trans %}Send feedback{% endtrans %}</button>
```
[the remark this answers is not in the document]
Comment (miguelahonen) [reply view/feedback.twig:48]: Good point! Aria-label removed.

File: resource/translations/skosmos_sv.po (hunk 1 above)
Comment (miguelahonen) [reply resource/translations/skosmos_sv.po:847]: Corrected

**Review by osma — APPROVED** [review]

**Comment by osma** [comment]
Great work, +1 for merging!

File: resource/css/styles.css [hunk 6]
```diff
@@ -5,6 +5,10 @@ li, td > a {
   font-size: 14px;
 }
 
+label {
```
Comment (kouralex) [inline resource/css/styles.css:8]: Global setting - this will affect all labels, in all places. See #799 for more information.

File: controller/WebController.php [hunk 7]
```diff
@@ -245,46 +243,43 @@ private function createFeedbackHeaders($fromName, $fromEmail, $toMail, $sender)
...
      */
-    public function sendFeedback($request, $message, $fromName = null, $fromEmail = null, $fromVocab = null, $toMail = null)
+    public function sendFeedback($request, $message, $messageSubject, $fromName = null, $fromEmail = null, $fromVocab = null, $toMail = null)
     {
         $toAddress = ($toMail) ? $toMail : $this->model->getConfig()->getFeedbackAddress();
+        $messageSubject = "[" . $this->model->getConfig()->getServiceName() . "] " . $messageSubject;
         if ($fromVocab !== null && $fromVocab !== '') {
             $message = 'Feedback from vocab: ' . strtoupper($fromVocab) . "<br />" . $message;
         }
-
```
Comment (kouralex) [inline controller/WebController.php:266]: Some of these lines may have been there for comprehensibility. It is not a good idea to remove them out of no reason. If one does want to do such operations, it is recommended to be done in a separate commit/branch in order to not confuse the PR.
